compiler/rustc_errors/src/lib.rs

```diff
@@ -1999,18 +1999,6 @@ pub fn a_or_an(s: &str) -> &'static str {
     }
 }
 
-/// Grammatical tool for displaying messages to end users in a nice form.
-///
-/// Take a list ["a", "b", "c"] and output a display friendly version "a, b and c"
-pub fn display_list_with_comma_and<T: std::fmt::Display>(v: &[T]) -> String {
-    match v {
-        [] => "".to_string(),
-        [a] => a.to_string(),
-        [a, b] => format!("{a} and {b}"),
-        [a, v @ ..] => format!("{a}, {}", display_list_with_comma_and(v)),
-    }
-}
```

Comment:
Removed this because it becomes redundant in the face of the new function, *and* it is less flexible (only works with pre-made lists that are displayable, but doesn't let you customize the format of each item, it is equivalent to `listify(v, |i| i.to_string())`).